### PR TITLE
Added IAM outputs, additional volumes mounts into runner, fix ssh var name

### PR DESCRIPTION
**Pull request by alexandrst88**
## Description
A few sentences describing the overall goals of the pull request's commits.
- Added option to mount additional volumes
- Added output for : runner_agent_role_arn, runner_role, runner_role_name, runner_role_arn
- Fix for ssh var name

## Migrations required
No

## Verification
- default and public

## Documentation
Please ensure you update the README in `_docs/README.md`. The README.md in the root can be update by running the script `ci/bin/autodocs.sh`

### CODE REVIEW

File: outputs.tf [hunk 1]
```diff
@@ -8,12 +8,27 @@ output "runner_cache_bucket_arn" {
...
+  value       = aws_iam_role.instance.name
+}
+
+output "runner_role_arn" {
   description = "ARN of the rule used for the docker machine runners."
   value       = aws_iam_role.docker_machine.arn
 }
+
+output "runner_role_name" {
+  description = "Name of the rule used for the docker machine runners."
```
Comment (npalm) [inline outputs.tf]: rule -> role (type)

File: outputs.tf [hunk 2]
```diff
@@ -8,12 +8,27 @@ output "runner_cache_bucket_arn" {
...
+}
+
+output "runner_agent_role_arn" {
   description = "ARN of the rule used for the ec2 instance for the GitLab runner agent."
   value       = aws_iam_role.instance.arn
 }
 
-output "runner_role" {
+output "runner_agent_role_name" {
+  description = "Name of the rule used for the ec2 instance for the GitLab runner agent."
```
Comment (npalm) [inline outputs.tf]: Typo rule -> role

File: outputs.tf [hunk 3]
```diff
@@ -8,12 +8,27 @@ output "runner_cache_bucket_arn" {
...
+}
+
+output "runner_role_arn" {
   description = "ARN of the rule used for the docker machine runners."
   value       = aws_iam_role.docker_machine.arn
 }
+
+output "runner_role_name" {
+  description = "Name of the rule used for the docker machine runners."
+  value       = aws_iam_role.docker_machine.arn
```
Comment (npalm) [inline outputs.tf]: The ARN is returned instead of the name. Name of variable not matching the return value

File: template/runner-config.tpl [hunk 4]
```diff
@@ -18,7 +18,7 @@ check_interval = 0
     image = "${runners_image}"
     privileged = ${runners_privileged}
     disable_cache = false
-    volumes = ["/cache"%{ if runners_mount_docker_socket == "true" },"${runners_docker_socket}"%{ endif ~}]
+    volumes = ["/cache"${runners_additional_volumes}]
```
Comment (npalm) [inline template/runner-config.tpl]: Thanks, I like the more generic approach.

File: variables.tf [hunk 5]
```diff
@@ -121,16 +122,10 @@ variable "runners_privileged" {
...
-}
-
-variable "runners_docker_socket" {
-  description = "Location of Docker socket on host if socket mount if enabled, will be used in the runner config.toml"
-  type        = "string"
-  default     = "/var/run/docker.sock:/var/run/docker.sock"
+variable "runners_additional_volumes" {
+  description = "Additional volumes that will be used in the runner config.toml, e.g Docker socket"
+  type        = list
+  default     = []
```
Comment (npalm) [inline variables.tf]: Maybe we could update the runner-public example to add an example usages

File: main.tf [hunk 6]
```diff
@@ -1,5 +1,5 @@
 resource "aws_key_pair" "key" {
-  count      = var.ssh_key_pair == "" ? 0 : 1
+  count      = var.ssh_public_key == "" ? 0 : 1
```
Comment (npalm) [inline main.tf]: `var.ssh_key_pair` was introduced to re-use existing keys. If set no key will be generated by the module. Your change looks like doing the same but for some reason the default example fails no.

```
Error: Invalid count argument

  on ../../main.tf line 2, in resource "aws_key_pair" "key":
   2:   count      = var.ssh_public_key == "" ? 0 : 1

The "count" value depends on resource attributes that cannot be determined
until apply, so Terraform cannot predict how many instances will be created.
To work around this, use the -target argument to first apply only the
resources that the count depends on.
```

Comment (npalm) [reply main.tf]: I solved the problem with key on in a slight other way.

Comment (hatemosphere) [reply main.tf]: @npalm so what we should leave in this PR and what should be removed to resolve the conflict?

**Comment by alexandrst88** [comment]
@npalm made the fixes that you requested, seems like build failed due to missed region

**Comment by hatemosphere** [comment]
@npalm hey, any chances to cut a new tag from the current branch state so we can move from our fork to upstream?

**Comment by npalm** [comment]
@hatemosphere new release 4.0.0_M4 - https://github.com/npalm/terraform-aws-gitlab-runner/releases/tag/4.0.0_M4

I hope to switch in the coming weeks by default to the TF 0.12 branch and release the 4.0.0 version.

**Pull request merged**